packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -0,0 +1,348 @@
+          E(zoeMintP).mintAndEscrow(totalToMint);
+          zcfSeatAdmin.commit(seatStaging);
+          return zcfSeat;
+        },
+        burnLosses: (losses, zcfSeat) => {
+          let totalToBurn = mintyAmountMath.getEmpty();
+          const oldAllocation = zcfSeat.getCurrentAllocation();
+          const updates = objectMap(
+            losses,
+            ([seatKeyword, amountToSubtract]) => {
+              totalToBurn = mintyAmountMath.add(totalToBurn, amountToSubtract);
+              const oldAmount = oldAllocation[seatKeyword];
+              const newAmount = mintyAmountMath.subtract(
+                oldAmount,
+                amountToSubtract,
+              );
+              return [seatKeyword, newAmount];
+            },
+          );
+          const newAllocation = harden({
+            ...oldAllocation,
+            ...updates,
+          });
+          const zcfSeatAdmin = seatToZCFSeatAdmin.get(zcfSeat);
+          // verifies offer safety
+          const seatStaging = zcfSeat.stage(newAllocation);
+          // No effects above. Commit point. The following two steps
+          // *should* be committed atomically.
+          // But unlike https://github.com/Agoric/agoric-sdk/issues/1391
+          // it is not a disater if they are not.
+          // If we only commit the staging, no one would ever get the
+          // unburned assets.
+          zcfSeatAdmin.commit(seatStaging);
+          E(zoeMintP).withdrawAndBurn(totalToBurn);
+        },
+      });
+      return zcfMint;
+    };
+
+    /** @type ContractFacet */
+    const zcf = {
+      reallocate: (/** @type SeatStaging[] */ ...seatStagings) => {
+        // We may want to handle this with static checking instead.
+        // Discussion at: https://github.com/Agoric/agoric-sdk/issues/1017
+        assert(
+          seatStagings.length >= 2,
+          details`reallocating must be done over two or more seats`,
+        );
+
+        seatStagings.forEach(seatStaging =>
+          assert(
+            allSeatStagings.has(seatStaging),
+            details`The seatStaging ${seatStaging} was not recognized`,
+          ),
+        );
+
+        // Ensure that rights are conserved overall. Offer safety was
+        // already checked when an allocation was staged for an individual seat.
+        const flattened = arr => [].concat(...arr);
+        const flattenAllocations = allocations =>
+          flattened(allocations.map(allocation => Object.values(allocation)));
+
+        const previousAllocations = seatStagings.map(seatStaging =>
+          seatStaging.getSeat().getCurrentAllocation(),
+        );
+        const previousAmounts = flattenAllocations(previousAllocations);
+
+        const newAllocations = seatStagings.map(seatStaging =>
+          seatStaging.getStagedAllocation(),
+        );
+        const newAmounts = flattenAllocations(newAllocations);
+
+        assert(
+          areRightsConserved(getAmountMath, previousAmounts, newAmounts),
```

Comment:
This is the only use of `areRightsConserved`. The above `flattenAllocations` and all the intermediary variables are only to set the stage for this one call to `areRightsConserved`. Should most of the logic above move into `areRightsConserved`?
```suggestion
          areRightsConserved(getAmountMath, previousAllocations, newAllocations),
```

Reply:
No, I believe it is good for areRightsConserved to have a clean, independent interface focused on calculating rights conservation. We might want to have a function that serves as a kind of adapter between the two kinds of representing the data.